File: RuntimeData/xnaruntime/RuntimeXNA/RuntimeXNA/Conditions/CND_NOSPSAMPLAYING.cs
// ------------------------------------------------------------------------------
// 
// NO SPECIFIC SAMPLE PLAYING?
// 
// ------------------------------------------------------------------------------
using System;
using RuntimeXNA.RunLoop;
using RuntimeXNA.Objects;
using RuntimeXNA.Params;
namespace RuntimeXNA.Conditions
{
	
	public class CND_NOSPSAMPLAYING:CCnd
	{
		public override bool eva1(CRun rhPtr, CObject hoPtr)
		{
			return eva2(rhPtr);
		}
		public override bool eva2(CRun rhPtr)
		{
			PARAM_SAMPLE p = (PARAM_SAMPLE) evtParams[0];
			return !rhPtr.rhApp.soundPlayer.isSamplePlaying(p.sndHandle);
		}
	}
}

File: RuntimeData/runtime/app/src/main/java/Conditions/CND_NOSPSAMPLAYING.java
// ------------------------------------------------------------------------------
// 
// NO SPECIFIC SAMPLE PLAYING?
// 
// ------------------------------------------------------------------------------
package Conditions;

import Objects.CObject;
import Params.CParam;
import Params.CParamExpression;
import Params.PARAM_SAMPLE;
import RunLoop.CRun;

public class CND_NOSPSAMPLAYING extends CCnd
{
    public boolean eva1(CRun rhPtr, CObject hoPtr)
    {
		return eva2(rhPtr);
    }
    public boolean eva2(CRun rhPtr)
    {
		CParam p = evtParams[0];
		boolean bPlaying = false;
		int nSound = -1;
	    // PARAM_EXPSTRING?
		if (p.code == 45) {
		    String name = rhPtr.get_EventExpressionString((CParamExpression)p);
		    nSound = rhPtr.rhApp.soundBank.getSoundHandleFromName(name);
		}
		else {
		    nSound = ((PARAM_SAMPLE)p).sndHandle;
		}
		if ( nSound >= 0 )
			bPlaying = rhPtr.rhApp.soundPlayer.isSamplePlaying((short)nSound);
		if (!bPlaying)
		{
			return negaTRUE();
		}
		return negaFALSE();
    }
}
